File: app/src/main/java/com/confessionsearch/release1/data/notes/InsertAsync.java
package com.confessionsearch.release1.data.notes;

import android.os.AsyncTask;

public class InsertAsync extends AsyncTask<Notes, Void, Void> {

    private final NoteDao mNoteDao;

    public InsertAsync(NoteDao dao) {
        mNoteDao = dao;
    }

    @Override
    protected Void doInBackground(Notes... notes) {
        mNoteDao.insertNotes(notes);
        return null;
    }}

File: app/src/main/java/com/confessionsearch/release1/data/notes/InsertAsync.kt
package com.confessionsearch.release1.data.notes


class InsertAsync(private val mNotesDao: NoteDao) :
    CoroutinesAsyncTask<Notes?, Void?, Void?>("insert")
{
    override fun doInBackground(vararg notes: Notes?): Void?
    {
        mNotesDao.insertNotes(*notes)
        return null
    }
}